### PR TITLE
[MOD-101][MOD-103][Fix] Add forbidden page and check permissions

### PR DESCRIPTION
**Pull request by laurenbarker**
### Tickets
https://openscience.atlassian.net/browse/MOD-101
https://openscience.atlassian.net/browse/MOD-103

### Purpose
Don't allow people to see pages they don't have permissions to see.

### Changes
* `/setup` will show `Not logged in` or `Forbidden` when appropriate (`set_up_moderation`)
* `/provider/*` routes will show `Not logged in` or `Forbidden` when appropriate (`canViewReviews`)

### CODE REVIEW

File: app/routes/preprints/provider.js [hunk 1]
```diff
@@ -10,6 +10,19 @@ import Ember from 'ember';
...
 export default Ember.Route.extend({
     theme: Ember.inject.service(),
+    session: Ember.inject.service(),
+    currentUser: Ember.inject.service(),
+
+    beforeModel() {
+        if (!this.get('session.isAuthenticated')) {
+            this.replaceWith('not-authenticated');
+        }
+        this.get('currentUser.user').then((user) => {
```
Comment (aaxelb) [inline app/routes/preprints/provider.js]: Maybe wrap this in `else`? I don't think `replaceWith` stops the flow of execution.

**Review by aaxelb — DISMISSED** [review]
Tiny nitpick, otherwise :squid:

File: app/routes/preprints/provider.js (hunk 1 above)
Comment (laurenbarker) [reply app/routes/preprints/provider.js]: Good idea. I ran into that on the other check but didn't hit it with this one for some reason.

**Review by aaxelb — APPROVED** [review]

**Comment by laurenbarker** [comment]
@aaxelb, done 💰

**Pull request merged**